### ISSUE TITLE
Intermittent failure of /_mozilla/css/nth_last_child_pseudo_a.html

### ISSUE THREAD

**Issue by KiChjang** [servo/servo#9699]
```
Ran 4600 tests finished in 228.0 seconds.
  • 4598 ran as expected. 1372 tests skipped.
  • 2 tests failed unexpectedly

Tests with unexpected results:
  ▶ FAIL [expected PASS] /_mozilla/css/nth_last_child_pseudo_a.html
  └   → /_mozilla/css/nth_last_child_pseudo_a.html 50508a1c9eaa256fa035416a57f3c052c032a74c
/_mozilla/css/nth_last_child_pseudo_b.html 3fdd75e0f66e51daa94b3112a8c2ce5ae5e37b24
Testing 50508a1c9eaa256fa035416a57f3c052c032a74c == 3fdd75e0f66e51daa94b3112a8c2ce5ae5e37b24
```

**Comment by mbrubeck** [comment]
May have a similar cause to #8191 (pseudo-class styles not being recalculated correctly on tree changes).

**Comment by jdm** [comment]
Maybe #9063 too!

**Comment by mbrubeck** [comment]
Should be fixed by #9722.
